pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -82,10 +83,14 @@ func retryOnConflictError(kubeClient clientset.Interface, namespace *api.Namespa
 		if !errors.IsConflict(err) {
 			return nil, err
 		}
+		prevNamespace := latestNamespace
 		latestNamespace, err = kubeClient.Core().Namespaces().Get(latestNamespace.Name)
 		if err != nil {
 			return nil, err
 		}
+		if prevNamespace.UID != latestNamespace.UID {
```

Comment:
this could happen if the retrieved namespace is the new one, or if the retrieved namespace is the old one... without interpreting resourceVersion, there's no way to know which it is. If we return an error here, does the cleanup get retried later?